tls/s2n_kem.c

```diff
@@ -238,14 +257,22 @@ int s2n_choose_kem_without_peer_pref_list(const uint8_t iana_value[S2N_TLS_CIPHE
 
 int s2n_kem_free(struct s2n_kem_params *kem_params)
 {
-    if (kem_params != NULL){
+    if (kem_params != NULL) {
         GUARD(s2n_blob_zeroize_free(&kem_params->private_key));
         GUARD(s2n_blob_zeroize_free(&kem_params->public_key));
         GUARD(s2n_blob_zeroize_free(&kem_params->shared_secret));
     }
     return S2N_SUCCESS;
 }
 
+int s2n_kem_group_free(struct s2n_kem_group_params *kem_group_params) {
+    if (kem_group_params != NULL) {
+        GUARD(s2n_kem_free(&kem_group_params->kem_params));
+        GUARD(s2n_ecc_evp_params_free(&kem_group_params->ecc_params));
```

Comment:
Not a trick question: if the kem fails will that cause the ecc to leak? Is that okay? Does the higher level connection cleanup double check all this?

Reply:
Not a trick answer: 
* "If the kem fails will that cause the ecc to leak?": Yes, if the `kem_free()` fails, then we probably do leak the ECC key.
* "Does the higher level connection cleanup double check all this?": It doesn't appear so based on what's currently in `s2n_connection_wipe_keys()` (and I did not add any additional checks in this PR): https://github.com/awslabs/s2n/blob/9f3f6fa1934adde7eca0ad0f22c0557994438eb7/tls/s2n_connection.c#L279-L308
* "Is that okay?": I think there is precedent for it being okay. Based on the above `s2n_connection_wipe_keys()` function, we hit a failure on, say, line 287, when wiping the client key, then we will leak the server key (along with everything that was supposed to be wiped on the subsequent lines).